api-admin/src/main/resources/api/openapi.yml

```diff
@@ -476,6 +476,26 @@ paths:
           content: { application/json: { schema: { type: object } } }
         '500':
           $ref: '#/components/responses/ServerError'
+  /api/portals/v1/{portalShortcode}/studies/{studyShortcode}/env/{envName}/notificationConfigs/{configId}:
+    patch:
```

Comment:
I'm not sure that `PATCH` is the right verb given that `configId` is a UUID and the effect is to create a new entity with a new UUID. I guess the alternative would be a `POST` to a generic update endpoint. Neither is really ideal, though I'm maybe 51% in favor of `POST` because it avoids me potentially making the wrong assumption about it's behavior when I see `PATCH`.

Reply:
Technically, [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) might be most appropriate.

Reply:
`PUT` should be idempotent. The client would have to choose the UUID `configId` to make it idempotent.

Reply:
I agree there's no clean answer to this -- the technical  operation is PATCH + POST, but we want to do them together because we don't want the study to not have a config if the PATCH succeeds but the POST fails.  And from a behavior perspective, PUT is also appropriate `replaces a representation of the target resource with the request payload`. Maybe I'll go with PUT since that's less common and so will clue the consumer that this is a complex operation

Reply:
On second thought, reading the docs on PUT, Brian, you're right they emphasize idempotency.  So I'm going to stick with PATCH, since that's the operation that's happening to the config identified by id in the URL (it's having it's `active` field set to false).